src/pe_reports/report_generator.py

```diff
@@ -28,39 +21,223 @@
+    cred_xlsx,
+    da_xlsx,
+    vuln_xlsx,
+    mi_xlsx,
+):
+    """Embeds raw data into PDF and encrypts file."""
+    doc = fitz.open(file)
+    # Get the summary page of the PDF on page 4
+    page = doc[3]
+    output = (
+        f"{output_directory}/{org_code}/Posture_and_Exposure_Report-{datestring}.pdf"
+    )
 
-def load_template():
-    """Load PowerPoint template into memory."""
-    prs = Presentation(REPORT_SHELL)
-    return prs
+    # Open CSV data as binary
+    cc = open(cred_xlsx, "rb").read()
+    da = open(da_xlsx, "rb").read()
+    ma = open(vuln_xlsx, "rb").read()
+    mi = open(mi_xlsx, "rb").read()
 
+    # Insert link to CSV data in summary page of PDF.
+    # Use coordinates to position them on the bottom.
+    p1 = fitz.Point(110, 695)
+    p2 = fitz.Point(240, 695)
+    p3 = fitz.Point(375, 695)
+    p5 = fitz.Point(500, 695)
 
-def export_set(prs, out_dir):
-    """Export PowerPoint report set to output directory."""
-    try:
-        prs.save(os.path.join(out_dir, REPORT_OUT))
-    except FileNotFoundError as not_found:
-        logging.error("%s : Missing input data. No report generated.", not_found)
+    # Embed and add push-pin graphic
+    page.add_file_annot(
+        p1, cc, "compromised_credentials.xlsx", desc="Open up CSV", icon="PushPin"
+    )
+    page.add_file_annot(
+        p2, da, "domain_alerts.xlsx", desc="Open up CSV", icon="PushPin"
+    )
+    page.add_file_annot(p3, ma, "vuln_alerts.xlsx", desc="Open up xlsx", icon="PushPin")
+    page.add_file_annot(
+        p5, mi, "mention_incidents.xlsx", desc="Open up CSV", icon="PushPin"
+    )
+
+    # Save doc and set garbage=4 to reduce PDF size using all 4 methods:
+    # Remove unused objects, compact xref table, merge duplicate objects,
+    # and check stream objects for duplication
+    doc.save(
+        output,
+        garbage=4,
+        deflate=True,
+    )
+    tooLarge = False
+    # Throw error if file size is greater than 20MB
+    filesize = os.path.getsize(output)
+    if filesize >= 20000000:
+        tooLarge = True
+
+    return filesize, tooLarge
+
+
+def convert_html_to_pdf(source_html, output_filename):
+    """Convert HTML to PDF."""
+    # Open output file for writing (truncated binary)
+    result_file = open(output_filename, "w+b")
+
+    # Convert HTML to PDF
+    pisa_status = pisa.CreatePDF(
+        source_html, dest=result_file  # the HTML to convert
+    )  # file handle to receive result
+
+    # Close output file
+    result_file.close()  # close output file
+
+    # Return False on success and True on errors
+    return pisa_status.err
+
+
+def generate_reports(datestring, output_directory):
+    """Process steps for generating report data."""
+    # Get PE orgs from PE db
+    conn = connect()
+    if conn:
+        pe_orgs = get_orgs(conn)
+    else:
+        return 1
+    generated_reports = 0
+
+    # Iterate over organizations
+    if pe_orgs:
+        logging.info("PE orgs count: %s", len(pe_orgs))
```

Comment:
Not a big deal, but since we know this one is a number:
```suggestion
        logging.info("PE orgs count: %d", len(pe_orgs))
```